syntax = "proto3";

package schema;

import "profanedb/protobuf/options.proto";

import "nested.proto";

message KeyInt {
  int32 int_key = 1 [ (profanedb.protobuf.options).key = true ];
}

message KeyStr {
  string string_key = 1 [ (profanedb.protobuf.options).key = true ];
}

message RepeatedKeyInt {
  repeated int32 int_key_repeated = 1 [ (profanedb.protobuf.options).key = true ];
}

message NonKeyableNested {
  int32 int_key = 1 [ (profanedb.protobuf.options).key = true ];
  
  message Nested {
    bool boolean = 1;
  }
  Nested nested_nonkeyable_message = 2;
}
